osutil/mountinfo_darwin.go

```diff
@@ -19,12 +19,6 @@
 
 package osutil
 
-import (
-	"io"
-	"os"
-)
-
-var openMountInfoFile = func() (io.ReadCloser, error) {
-	mustNotBeTestBinary("/proc/self/mountinfo must be mocked in tests")
-	return os.Open("/proc/self/mountinfo")
+func MockMountInfo(content string) (restore func()) {
+	return func() {}
```

Comment:
hmm this is a bit weird in terms of mocking behavior, the unit tests when run on mac will fail for weird reasons since they call the MockMountInfo function and expect it to setup mocking, then when it doesn't setup mocking it's unclear how they will fail.

I had missing in my original review that this is the function that needs to be mocked, I thought it would be the other one... 

In any case can you adjust this to just panic on darwin so tests on darwin at least fail in an expected way and not in a weird way in case anyone ever tries to run/debug unit tests on darwin:



```suggestion
func MockMountInfo(content string) (restore func()) {
	panic("MockMountInfo not implemented on darwin")
```

Reply:
@anonymouse64, thank you for catching this one.
By the way, you couldn't see it during your original review, since the code structure was a bit different. I was asked to remove Linux logic out of common code, which resulted in some refactoring. This refactoring produced this instance of the mocking API. Before it, Darwin didn't have it's own version.